copy.go

```diff
@@ -87,7 +89,13 @@ func fcopy(src, dest string, info os.FileInfo, opt Options) (err error) {
 	}
 	defer fclose(s, &err)
 
-	if _, err = io.Copy(f, s); err != nil {
+	if opt.CopyBufferSize <= 0 {
+		_, err = io.Copy(f, s)
+	} else {
+		buf := make([]byte, opt.CopyBufferSize)
+		_, err = io.CopyBuffer(onlyWriter{f}, s, buf)
```

Comment:
Wrapping `onlyWriter{f}` is necessary, because else the specified buffer is never used internally, because the file writer complies to the `io.ReaderFrom` interface which has the effect, that the `CopyBuffer` function not used the buffer but instead forwards the request to the `io.ReaderFrom`. This `ReaderFrom` is sometimes a good choice but in case it can't handle the request (e.g. too old kernel) it falls back to the previous copy function but without any buffer (except the default 32K): See


Excerpt of `file.go`
```
// ReadFrom implements io.ReaderFrom.
func (f *File) ReadFrom(r io.Reader) (n int64, err error) {
	if err := f.checkValid("write"); err != nil {
		return 0, err
	}
	n, handled, e := f.readFrom(r)
	if !handled {
		return genericReadFrom(f, r) // without wrapping
	}
	return n, f.wrapErr("write", e)
}
```